http_client.rst

```diff
@@ -883,21 +883,18 @@ In case the response was canceled using ``$response->cancel()``,
 Handling Exceptions
 ~~~~~~~~~~~~~~~~~~~
 
-When the HTTP status code of the response is in the 300-599 range (i.e. 3xx,
-4xx or 5xx) your code is expected to handle it. If you don't do that, the
-``getHeaders()`` and ``getContent()`` methods throw an appropriate exception, all of
-which implement the :class:`Symfony\\Contracts\\HttpClient\\Exception\\HttpExceptionInterface`::
+If the HTTP status code of the response is in the 300-599 range (i.e. 3xx, 4xx or 5xx),
+the ``getHeaders()`` and ``getContent()`` methods throw an appropriate
+exception, which implements the
+:class:`Symfony\\Contracts\\HttpClient\\Exception\\HttpExceptionInterface`
 
-    // the response of this request will be a 403 HTTP error
-    $response = $client->request('GET', 'https://httpbin.org/status/403');
+The preferred way to handle this is to call ``$response->getStatusCode()``, cause this
+will prevent subsequent ``getHeaders()`` and ``getContent()`` calls from throwing an
```

Comment:
Let's start with the easy question: I guess that ->toArray() behaves the same as getHeaders() and getContent()? So it should me mentioned too?

Now the hard: So you're saying that `getHeaders(true)` throws always? But it's not the destructor but some other code part that is issuing the exception (which doesn't make a difference to the user)? So which difference does calling `getStatusCode()` then make at all?

Sorry, but every time I think I've got it, I realize that I haven't ;-)

Reply:
Yes, toArray also has the `$throw` argument
All the methods that have this argument throw accordingly, whether you called `getStatusCode()` or not.
On destruction, if you didn't call `getStatusCode()`, an exception will be thrown when the status is >= 300.

Reply:
Sorry, I still don't get it :-( Here's the code sample from https://symfony.com/doc/4.4/http_client.html#handling-exceptions
```php
// the response of this request will be a 403 HTTP error
$response = $client->request('GET', 'https://httpbin.org/status/403');

// this code results in a Symfony\Component\HttpClient\Exception\ClientException
// because it doesn't check the status code of the response
$content = $response->getContent();
```
Could you please show me the full code that "because it doesn't check the status code of the response" refers to?